hazelcast/src/main/java/com/hazelcast/map/impl/PartitionContainer.java

```diff
@@ -122,6 +124,22 @@ private RecordStore createRecordStore(String name) {
         return maps.values();
     }
 
+    public Collection<ServiceNamespace> getAllNamespaces(int replicaIndex) {
+        Collection<ServiceNamespace> namespaces = new HashSet<ServiceNamespace>();
+
+        for (RecordStore recordStore : maps.values()) {
```

Comment:
seems using `mapServiceContext.getMapContainers()` would be better to eliminate excessive looping on record-stores